submissions/pozzitive11/friends-app/script.js

```diff
@@ -0,0 +1,153 @@
+const filterGenderBlock = document.querySelector(".sort-gender");
+
+const searchInput = document.querySelector("#search-input");
+const resetBtn = document.querySelector(".form__button");
+
+const burgerBtn = document.querySelector(".burger");
+const sidebar = document.querySelector(".sidebar");
+
+burgerBtn.addEventListener("click", (e) => {
+  burgerBtn.classList.toggle("active");
+  sidebar.classList.toggle("sidebar--visible");
+});
+
+const statusMessage = {
+  loading: "img/spinner.svg",
+  failure: "Technical problems, try again later",
+};
+
+const loadingMessage = document.createElement("img");
+function showLoadingStatus() {
+  loadingMessage.src = statusMessage.loading;
+  loadingMessage.classList.add("status-block");
+
+  cardsList.insertAdjacentElement("afterend", loadingMessage);
+}
+
+function loadUsersData() {
+  showLoadingStatus();
+
+  fetch(
+    "https://randomuser.me/api/?results=20&nat=us,ua,de&inc=picture,name,dob,gender,location"
+  )
+    .then((response) => {
+      if (response.ok) return response.json();
+    })
+    .then((responseJson) => {
+      users = responseJson.results;
+      showCards(users);
+      loadingMessage.remove();
+    })
+    .catch(() => {
+      loadingMessage.remove();
+      showErrorStautus();
+    });
+}
+
+function showErrorStautus() {
+  const errorMessage = document.createElement("div");
+  errorMessage.classList.add("error-block");
+  errorMessage.innerHTML = `${statusMessage.failure}`;
+
+  console.log(statusMessage.failure);
+  cardsList.append(errorMessage);
+}
+
+function showCards(cards) {
+  cards.forEach(({ picture, name, dob, gender, location }) => {
+    createCard({ picture, name, dob, gender, location });
+  });
+}
+
+function createCard({ picture, name, dob, gender, location }) {
+  const userCard = document.createElement("li");
+  userCard.classList.add("user__item");
+  let userGender;
+  if (gender === "male") userGender = "user__gender--male";
+  if (gender === "female") userGender = "user__gender--female";
+  userCard.innerHTML = `
+    <h4 class="user__gender ${userGender}">${gender}</h4>
+    <img class="user__img" src="${picture.large}" alt="User photo" />
+    <p class="user__name">${name.first} ${name.last}</p>
+    <p class="user__age">Age: <span class="user__age-span">${dob.age}</span></p>
+    <p class="user__location">${location.country}</p>`;
+
+  cardsList.append(userCard);
+}
+
+const compareByName = (firstUser, secondUser) =>
+  firstUser.name.first.toLowerCase() <= secondUser.name.first.toLowerCase()
+    ? -1
+    : 1;
+const compareByAge = (firstUser, secondUser) =>
+  firstUser.dob.age - secondUser.dob.age;
+
+const compareByGender = (user, type) => user.gender === type;
+
+function filterBySearch(arr, target) {
+  return arr.filter((user) => {
+    const fullName = `${user.name.first} ${user.name.last}`.toLowerCase();
+    return fullName.includes(target.toLowerCase());
+  });
+}
+
+function filterUsers({ target }) {
```

Comment:
This function should modify users data and return result. Single responsibility principle